devito/types/dimension.py

```diff
@@ -718,7 +720,17 @@ def __init_finalize__(self, name, parent, factor=None, condition=None,
         # import pdb; pdb.set_trace()
         processed = []
         expr = condition
-        dimension_map = {}
+
+        # Awful TOFIX
```

Comment:
NOTE FOR REVIEWERS:
We intend to support subdomains for ConditionalDimensions's conditions.
I am trying to migrate extended sympy's Classes to `equation.py`
 and inherit subdomain properties. 
Pure Sympy conditions like `And` that is heavily used in injection and interpolation do not yet support a subdomain
in their expression.
That's why the awful hack here that needs to be fixed.
As said in previous comment the code below is duplicate to indexify_expr.